modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/processor/AbstractSQLProcessor.java

```diff
@@ -325,17 +327,29 @@ public ODataResponse createEntity(final PostUriInfo uriInfo, final InputStream c
         ODataEntry entry = parseEntry(entitySet, content, requestContentType, false);
 
         Map<Object, Object> handlerContext = new HashMap<>();
-        Map<String, Object> entryMap = null;
+        ODataResponse response;
         try (Connection connection = getDataSource().getConnection()) {
             updateCreateEventHandlerContext(handlerContext, connection, uriInfo, entry);
-            this.odata2EventHandler.beforeCreateEntity(uriInfo,
+            response = this.odata2EventHandler.beforeCreateEntity(uriInfo,
                     requestContentType, contentType, entry, handlerContext);
+            if (isErroneousResponse(response)) {
+                return response;
+            }
 
             if (this.odata2EventHandler.usingOnCreateEntity(uriInfo, requestContentType, contentType)) {
                 updateCreateEventHandlerContext(handlerContext, connection, uriInfo, entry);
-                this.odata2EventHandler.onCreateEntity(uriInfo, content,
+                response = this.odata2EventHandler.onCreateEntity(uriInfo, content,
                         requestContentType, contentType, handlerContext);
-                entryMap = (HashMap<String, Object>) handlerContext.get(ENTRY_MAP);
+                if (isErroneousResponse(response)) {
+                    return response;
+                } else if (response != null) {
+                    ByteArrayOutputStream entityOutputStream = new ByteArrayOutputStream();
+                    ((InputStream) response.getEntity()).transferTo(entityOutputStream);
+                    InputStream entityInputStream = new ByteArrayInputStream(entityOutputStream.toByteArray());
+                    InputStream entityCloneInputStream = new ByteArrayInputStream(entityOutputStream.toByteArray());
+                    entry = parseEntry(entitySet, entityInputStream, response.getContentHeader(), false);

```

Comment:
Do you need this logic? Why not just return the response or ODataResponse.fromResponse(response)? And all streams must be closed.